src/Altinn.Platform/Altinn.Platform.Events/Events/Controllers/SubscriptionController.cs

```diff
@@ -0,0 +1,306 @@
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private async Task<string> GetPartyFromAlternativeSubject(string alternativeSubject)
+        {
+            int partyId = 0;
+
+            if (alternativeSubject.StartsWith(OrganizationPrefix))
+            {
+                string orgNo = alternativeSubject.Replace(OrganizationPrefix, string.Empty);
+                partyId = await _registerService.PartyLookup(orgNo, null);
+            }
+            else if (alternativeSubject.StartsWith(PersonPrefix))
+            {
+                string persnoNo = alternativeSubject.Replace(PersonPrefix, string.Empty);
+                partyId = await _registerService.PartyLookup(null, persnoNo);
+            }
+
+            if (partyId != 0)
+            {
+                return "/party/" + partyId;
+            }
+
+            return null;
+        }
+
+        private void EnrichConsumer(EventsSubscription eventsSubscription)
+        {
+            if (string.IsNullOrEmpty(eventsSubscription.Consumer))
+            {
+                string org = HttpContext.User.GetOrg();
+                if (!string.IsNullOrEmpty(org))
+                {
+                    eventsSubscription.Consumer = "/org/" + org;
+                    return;
+                }
+
+                int? userId = HttpContext.User.GetUserIdAsInt();
+                if (userId.HasValue)
+                {
+                    eventsSubscription.Consumer = "/user/" + userId.Value;
+                    return;
+                }
+
+                string organization = HttpContext.User.GetOrgNumber();
+                if (!string.IsNullOrEmpty(organization))
+                {
+                    eventsSubscription.Consumer = "/organization/" + organization;
+                    return;
+                }
+            }
+        }
+
+        private void SetCreatedBy(EventsSubscription eventsSubscription)
+        {
+            string org = HttpContext.User.GetOrg();
+            if (!string.IsNullOrEmpty(org))
+            {
+                eventsSubscription.CreatedBy = "/org/" + org;
+                return;
+            }
+
+            int? userId = HttpContext.User.GetUserIdAsInt();
+            if (userId.HasValue)
+            {
+                eventsSubscription.CreatedBy = "/user/" + userId.Value;
+                return;
+            }
+
+            string organization = HttpContext.User.GetOrgNumber();
+            if (!string.IsNullOrEmpty(organization))
+            {
+                eventsSubscription.CreatedBy = "/organization/" + organization;
+                return;
+            }
+        }
+
+        private bool AuthorizeAccessToSubscription(EventsSubscription eventsSubscription)
+        {
+            string currentIdenity = string.Empty;
+
+            if (!string.IsNullOrEmpty(HttpContext.User.GetOrg()))
+            {
+                currentIdenity = "/org/" + HttpContext.User.GetOrg();
```

Comment:
use the defined prefix const for all three cases here.

Reply:
updated